### DISCUSSION TITLE
Issues with Kotlin

### DISCUSSION THREAD

**Discussion by Rychillie** [skiptools/skip#13] [Q&A] [answered]
I'm running `skip checkup` and returning an error in `Test Kotlin`(at least what I want to solve at the moment), I have installed Xcode 15 and Android Studio in its latest version and with Kotlin in its stable version, I imagine it could be something which I forgot to install, because my MacBook is new and I haven't fully configured it yet.

```bash
❯ skip checkup
[✓] Skip version 0.7.13 (= 0.7.13)
[✓] macOS version 14.1 (> 13.5.0)
[✓] Swift version 5.9 (= 5.9.0)
[✓] Xcode version 15.0.1 (> 15.0.0)
[✓] Homebrew version 4.1.17 (> 4.1.0)
[✓] Gradle version 8.4 (> 8.3.0)
[✓] Java version 21.0.1 (> 17.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.3
[✓] Create project hello-skip (0.37s)
[✓] Resolve dependencies (11.28s)
[✓] Build hello-skip (15.94s)
[✓] Test Swift (10.03s)
[✗] Test Kotlin (2.44s)
[✓] Archive iOS ipa (17.42s)
[✓] Assemble HelloSkip-release.ipa (0.01s)
[✓] Verify HelloSkip-release.ipa 23 KB
[✗] Assembling Android apk (0.64s)
[✗] Verify HelloSkip-release.apk: The file “HelloSkip-release-unsigned.apk” couldn’t be opened because there is no such file.
[✗] Skip 0.7.13 checkup (59.18s)
Error: 4 errors
```

**Comment by marcprux** [comment, marked as answer]
It looks like you have Java 21 installed. There's a known issue with Java 21 and Gradle, as discussed at https://github.com/skiptools/skip/issues/5. Skip should have installed `openjdk@17` as part of the Homebrew installation, but it doesn't look like it is being used by your install (from looking at the `Java version 21.0.1 (> 17.0.0)` output of `skip doctor`).

You might be able to just clear your `JAVA_HOME` and re-run; Gradle should they try to use the preferred JDK. Can you try running: `JAVA_HOME= skip checkup` and see if it works? If not, can you post the output from `JAVA_HOME= skip checkout --verbose`?

**Reply by Rychillie** [reply]
Fixed with issue #5 process
